### ISSUE TITLE
Metric fluctuations

### ISSUE THREAD

**Issue by linxuewu** [HorizonRobotics/Sparse4D#8]
There is some significant fluctuation in the metrics of the nuScenes dataset, and some users are experiencing considerable differences in the reproduced metrics compared to the ones we have publicly shared. Therefore, I conducted multiple sets of repeated experiments and provided a range for the metrics as a reference for everyone.

**The conclusion shows that NDS is a relatively stable metric.**

| |NDS|AMOTA|mAP|mATE|mASE|mAOE|mAVE|mAAE|AMOTP|RECALL|IDS|
|:----|:----|:----|:----|:----|:----|:----|:----|:----|:----|:----|:----|
|released|0.5636|0.477|0.4647|0.5403|0.2623|0.459|0.2198|0.2059|1.136|0.588|441|
|1|0.5608|0.476|0.4597|0.5561|0.2641|0.4573|0.2166|0.1959|1.152|0.568|418|
|2|0.5657|0.487|0.4702|0.5358|0.2652|0.4781|0.2162|0.1988|1.135|0.585|463|
|3|0.5647|0.465|0.4692|0.5256|0.2617|0.5031|0.2186|0.1898|1.133|0.566|431|
|4|0.5607|0.457|0.4512|0.5382|0.2661|0.434|0.2203|0.191|1.154|0.567|439|
|5|0.5634|0.466|0.4585|0.5575|0.2664|0.4157|0.2149|0.2038|1.165|0.581|510|
|6|0.5683|0.474|0.469|0.5259|0.2646|0.4571|0.2139|0.2003|1.138|0.568|389|
|mean|0.5639|0.4717|0.4632|0.5399|0.2643|0.4578|0.2172|0.1979|1.1447|0.5747|441.5714|
|std|0.0027|0.0098|0.0071|0.0129|0.0018|0.0283|0.0024|0.0061|0.0123|0.0096|37.8147|

**Comment by linxuewu** [comment]
If sparse4d is trained with a smaller batch_size, the learning rate should be lower. For example, if you take `num_gpus * bs_per_gpu = 4 * 6 = 24`, the learning rate could be set as 4e-4. Below are the experimental results under this configuration.

|NDS|AMOTA|mAP|mATE|mASE|mAOE|mAVE|mAAE|AMOTP|RECALL|IDS|
|:----|:----|:----|:----|:----|:----|:----|:----|:----|:----|:----|
|0.5675|0.471|0.4592|0.5276|0.2614|0.4185|0.2241|0.1890|1.139|0.570|398|

**Comment by ASONG0506** [comment]
![image](https://github.com/HorizonRobotics/Sparse4D/assets/26050577/a07fdd11-5ee7-43f6-832b-3c410541bbb5)
My result using 4*6=24 total_batchsize, lr=4e-4. seems like normal

**Comment by tiger-sheep** [comment]
您好，我用两块GPU（RTX TiTAN），batch_size 为 2*6=12，学习率设置为2e-4，训练出来mAP只有0.26。能给一点调参的建议吗？万分感谢！

**Comment by linxuewu** [comment]
> 您好，我用两块GPU（RTX TiTAN），batch_size 为 2*6=12，学习率设置为2e-4，训练出来mAP只有0.26。能给一点调参的建议吗？万分感谢！

log发出来看看

**Comment by tiger-sheep** [comment]
> > 您好，我用两块GPU（RTX TiTAN），batch_size 为 2*6=12，学习率设置为2e-4，训练出来mAP只有0.26。能给一点调参的建议吗？万分感谢！
> 
> log发出来看看

![屏幕截图 2024-04-26 173251](https://github.com/HorizonRobotics/Sparse4D/assets/74152717/6d0e1182-21a1-4bf7-956f-145a47a49197)

**Comment by tiger-sheep** [comment]
![屏幕截图 2024-04-26 173251](https://github.com/HorizonRobotics/Sparse4D/assets/74152717/f0d67a57-3410-434d-9d17-3bd709419279)
![屏幕截图 2024-04-26 173511](https://github.com/HorizonRobotics/Sparse4D/assets/74152717/e5fb9fb5-4345-4473-9390-4bdc47de638d)
[20240425_110113.log](https://github.com/HorizonRobotics/Sparse4D/files/15128453/20240425_110113.log)


> > 您好，我用两块GPU（RTX TiTAN），batch_size 为 2*6=12，学习率设置为2e-4，训练出来mAP只有0.26。能给一点调参的建议吗？万分感谢！
> 
> log发出来看看

**Comment by linxuewu** [comment]
1. 看看代码是不是更新到最新版本了，前段时间我修了一个bug，[commit link](https://github.com/HorizonRobotics/Sparse4D/commit/2543c77d5c55bee71676a1326bdf2fc6deefbccb)；
2. 尝试更小的学习率，比如1e-4，以及把`img_backbone=dict(lr_mult=0.2)`等；
3. 如果还不行的话，建议把数据增强关了。

**Comment by tiger-sheep** [comment]
> 1. 看看代码是不是更新到最新版本了，前段时间我修了一个bug，[commit link](https://github.com/HorizonRobotics/Sparse4D/commit/2543c77d5c55bee71676a1326bdf2fc6deefbccb)；
> 2. 尝试更小的学习率，比如1e-4，以及把`img_backbone=dict(lr_mult=0.2)`等；
> 3. 如果还不行的话，建议把数据增强关了。

我根据您的建议修改了参数配置，训练结果很正常了。再次感谢！
